components/lib/hooks/useWeb3.tsx

```diff
@@ -57,28 +57,41 @@ export const UseWeb3Provider: React.FC<ChildrenShim> = ({ children }) => {
     }
   }, []);
 
+  const idToHexString = (id: number) => {
+    return "0x" + id.toString(16);
+  };
+
   async function connectMetamask() {
     if (metamaskInstalled) {
       const newProvider = new ethers.providers.Web3Provider(window.ethereum);
-      setWeb3State({ ...web3State, connecting: true });
-      const addresses = (await newProvider.send("eth_requestAccounts", [])) as string[];
-      if (addresses.length > 0) {
-        console.log("Connected wallet ", addresses[0]);
-        const newWalletAddress = addresses[0];
-        const newSigner = newProvider.getSigner(newWalletAddress);
-        setStoredWallet(newWalletAddress);
-        setStoredProviderMode("metamask");
-        setWeb3State({
-          ...web3State,
-          connecting: false,
-          providerMode: "metamask",
-          provider: newProvider,
-          walletAddress: newWalletAddress,
-          signer: newSigner,
-        });
+
+      const chainId = await newProvider.getNetwork().then((network) => network.chainId);
+      if (chainId === 1) {
```

Comment:
For hardcoded chainId, could we update it just like read it from .env? because the dapp should be able to work on any networks we deploy smart contracts on.

Reply:
Yes, I can make it work from reading the .env and remove the hardcoded part.